src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -53,103 +68,844 @@ static ColliderJntSphInit D_809BF3F8 = {
+        if (this->unk_196 == -10) {
+            func_809BD3AC(this);
+        }
+    }
+}
+
+void func_809BDAE8(EnBigokuta* this, GlobalContext* globalCtx) {
+    if (Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y + 0x4000, 0x400)) {
+        if (this->unk_196 != 0) {
+            this->unk_196--;
+        }
+        if (this->unk_196 == 0) {
+            func_809BCE3C(this);
+            this->actor.home.pos.y = this->actor.world.pos.y;
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_ENEMY);
+            this->actor.params = 2;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_VOICE);
+            func_809BD3E0(this);
+        }
+    }
+}
+
+void func_809BDB90(EnBigokuta* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk_196 != 0) {
+        this->unk_196--;
+    }
+    if (this->unk_196 == 0) {
+        if (this->actor.params == 3) {
+            func_800F5ACC(0x38U);
+        }
+        func_809BD3F8(this);
+    }
+}
+
+void func_809BDC08(EnBigokuta* this, GlobalContext* globalCtx) {
+    Player* player;
+    s16 phi_v0;
+    s16 pad;
+    s16 phi_v1;
+    Vec3f sp28;
+
+    player = PLAYER;
+    SkelAnime_Update(&this->skelAnime);
+    if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_OCTAROCK_BUBLE);
+    }
+
+    if (this->unk_196 < 0) {
+        this->actor.shape.rot.y += this->unk_194 * 0x200;
+        func_809BCE3C(this);
+        this->unk_196++;
+        if (this->unk_196 == 0) {
+            this->unk_196 = 350;
+        }
+        func_809BCF68(this, globalCtx);
+        return;
+    }
+
+    phi_v1 = (Actor_WorldDistXZToPoint(&player->actor, &this->actor.home.pos) - 180.0f) * (8.0f / 15);
+    func_8002DBD0(&this->actor, &sp28, &player->actor.world.pos);
+    if (fabsf(sp28.x) > 263.0f || ((sp28.z > 0.0f) && !Actor_IsFacingPlayer(&this->actor, 0x1B00) &&
+                                   !Player_IsFacingActor(&this->actor, 0x2000, globalCtx))) {
+        phi_v1 -= 0x80;
+        if (this->unk_196 != 0) {
+            this->unk_196--;
+        }
+    }
+
+    if ((this->actor.xzDistToPlayer < 250.0f) && !Actor_IsFacingPlayer(&this->actor, 0x6000)) {
+        if (this->unk_198 != 0) {
+            this->unk_198--;
+        }
+        if (this->actor.xzDistToPlayer < 180.0f) {
+            phi_v1 += 0x20;
+        }
+    } else {
+        this->unk_198 = 80;
+    }
+    if (this->actor.colChkInfo.health == 1) {
+        phi_v1 = (phi_v1 + 0x130) * 1.1f;
+    } else {
+        phi_v1 += 0x130;
+    }
+    this->actor.shape.rot.y += phi_v1 * this->unk_194;
+    func_809BCE3C(this);
+    func_809BCF68(this, globalCtx);
+    if (this->unk_198 == 0) {
+        func_809BD768(this);
+        return;
+    }
+    if (this->unk_196 == 0) {
+        func_809BD4A4(this);
+        return;
+    }
+    if (this->unk_195) {
```

Comment:
```suggestion
    } else if (this->unk_196 == 0) {
        func_809BD4A4(this);
    } else if (this->unk_195) {
```
though function setups, no early returns looks cleaner imo (the blocks are already there anyway)